contracts/exchange-forwarder/contracts/src/MixinExchangeWrapper.sol

```diff
@@ -316,4 +316,40 @@ contract MixinExchangeWrapper is
             ));
         }
     }
+
+    /// @dev Checks whether an order's taker fee is effectively denominated in the maker asset.
+    ///      This is the case if they have the same ERC20Proxy asset data, or if the makerAssetData
+    ///      is the ERC20Bridge equivalent of the takerFeeAssetData.
+    /// @param takerFeeAssetData Byte array encoded for the takerFee asset proxy.
+    /// @param makerAssetData Byte array encoded for the maker asset proxy.
+    /// @return isPercentageFee Whether or not the taker fee asset matches the maker asset.
+    function _isPercentageFee(
+        bytes memory takerFeeAssetData,
+        bytes memory makerAssetData
+    )
+        internal
+        pure
+        returns (bool)
+    {
+        bytes4 takerFeeAssetProxyId = takerFeeAssetData.readBytes4(0);
+        // If the takerFee asset is not ERC20, it cannot be a percentage fee (and will revert with
+        // an UnsupportedFeeError in the calling function).
+        if (takerFeeAssetProxyId != ERC20_PROXY_ID) {
```

Comment:
There's probably some edge case where it's possible that the `takerFeeAssetProxyId == ERC20_BRIDGE_PROXY_ID`. It's probably simpler if we just check:

```
if (
    (takerFeeAssetProxyId == ERC20_PROXY_ID || takerFeeAssetProxyId == ERC20_PROXY_ID) &&
    (makerAssetProxyId == ERC20_PROXY_ID || makerAssetProxyId == ERC20_PROXY_ID)
) { ... }
```